test.js

```diff
@@ -329,6 +329,20 @@ var TESTS = [
     { param: '["a", 1, tru]' },
     { param: '["a", 1, tru]' }
   ],
+  [
+    {
+      param: { type: ['array'] }
+    },
+    { param: 123 },
+    { param: [123] }
```

Comment:
is this test case currently failing (without this PR)? If not, can you make sure to add as many test cases as this PR covers? would also be nice to have a case with an array of one string. Can it ever be an array of objects, if so let's an array of one object too.